datalad/cmd.py

```diff
@@ -353,6 +353,111 @@ def log(self, msg, level=logging.DEBUG):
             lgr.log(level, "{%s} %s" % (self.protocol.__class__.__name__, msg))
 
 
+class GitRunner(Runner):
+    """
+    Overloads the runner class to check update GIT_DIR and GIT_WORK_TREE environment variables set to the correct path
+    """
+
+    def run(self, cmd, log_stdout=True, log_stderr=True, log_online=False,
+            expect_stderr=False, expect_fail=False,
+            cwd=None, env=None, shell=None):
+
+        # assert_true(cwd, None)
+        # assert_false(self.cwd, None)
+        env = {"GIT_DIR": self.cwd+"/.git", "GIT_WORK_TREE": self.cwd+"/"}  # set environment variables, rm chdir error
```

Comment:
do not just override env variable 
- use it as a starting point (`adjusted_env = env.copy() if env else {}`)
- analize current os.environ if it contains GIT_DIR or GIT_WORK_TREE, and iff does, then adjust those values taking into account `getpwd` (don't use `os.getcwd`)... it might actually suffice to just `abspath` them ...

so altogether solution should be just 2-3 lines ;)

you might also like to not list all args/kwargs above explicitly but just use `*args, **kwargs`, and then get env via `kwargs.get(env, {})`